app/assets/javascripts/darkswarm/directives/validate_stock_quantity.js.coffee

```diff
@@ -0,0 +1,17 @@
+Darkswarm.directive "validateStockQuantity", ->
+  restrict: 'A'
+  require: "ngModel"
+
+  link: (scope, element, attr, ngModel) ->
+    ngModel.$parsers.push (selectedQuantity) ->
+      if parseInt(selectedQuantity) > scope.available_quantity()
+        ngModel.$setValidity('stock', false);
+      else
+        ngModel.$setValidity('stock', true);
+
+      selectedQuantity
+
+    scope.available_quantity = ->
+      on_hand = parseInt(attr.ofnOnHand)
+      finalized_quantity = parseInt(attr.finalizedquantity) || 0 # finalizedquantity is optional
+      on_hand + finalized_quantity
```

Comment:
I think this is complicated code already in ofnOnHand, I dont think we can copy paste this type of logic. It needs to be extracted to a service to be re-used here.

Reply:
I've made this more explicit :+1:

Reply:
Done.